src/application/utils/taxi.ts

```diff
@@ -1,24 +1,35 @@
-import { TaxiClient } from 'taxi-protobuf/generated/js/TaxiServiceClientPb';
-import {
-  AssetDetails,
-  ListAssetsRequest,
-  TopupWithAssetReply,
-  TopupWithAssetRequest,
-} from 'taxi-protobuf/generated/js/taxi_pb';
+import axios from 'axios';
+
+interface AssetDetails {
+  assetHash: string;
+  assetPrice: number;
+  basisPoint: number;
+}
+
+export interface Topup {
+  assetAmount: number;
+  assetHash: string;
+  assetSpread: number;
+  partial: string;
+  topupId: string;
+}
+
+export interface TopupWithAssetReply {
+  expiry: number;
+  privateBlindingKey: string;
+  publicBlindingKey: string;
+  topup?: Topup;
+}
 
 export const fetchAssetsFromTaxi = async (taxiUrl: string): Promise<string[]> => {
-  const client = new TaxiClient(taxiUrl, undefined);
-  const res = await client.listAssets(new ListAssetsRequest(), null);
-  return res.getAssetsList().map((asset: AssetDetails) => asset.getAssetHash());
+  const { data } = await axios.get(`${taxiUrl}/assets`);
+  return data.assets.map((asset: AssetDetails) => asset.assetHash);
 };
 
 export const fetchTopupFromTaxi = async (
   taxiUrl: string,
-  asset: string
-): Promise<TopupWithAssetReply.AsObject> => {
-  const client = new TaxiClient(taxiUrl, undefined);
-  const request = new TopupWithAssetRequest();
-  request.setAssetHash(asset);
-  const res = await client.topupWithAsset(request, null);
-  return res.toObject();
+  assetHash: string
+): Promise<TopupWithAssetReply> => {
+  const { data } = await axios.post(`${taxiUrl}/asset/topup`, assetHash);
```

Comment:
We miss third parameter, with an object and a key `headers` where we specify application/json

strange works for you

Reply:
It was working because I had `{ assetHash }` has second parameter, and when we have an object has second parameter axios adds the `content-type application/json` by default. My last commit introduced the error, was not properly tested. This new commit 9eaef50 fixes it. @tiero can you try it?